test/test_torch.py

```diff
@@ -3068,23 +3069,106 @@ def test_Size(self):
         self.assertIsInstance(x[:-1], torch.Size)
         self.assertIsInstance(x + x, torch.Size)
 
-    def test_transpose_neg(self):
-        x = torch.randn(10, 20, 30)
-        ndim = 3
 
-        for i, j in combinations(range(ndim), 2):
-            a = x.transpose(i, j)
-            b = x.transpose(i - ndim, j - ndim)
-            self.assertEqual(a, b)
-
-            a = torch.transpose(x, i, j)
-            b = torch.transpose(x, i - ndim, j - ndim)
-            self.assertEqual(a, b)
-
-            a = x.clone()
-            x.transpose_(i, j)
-            x.transpose_(i - ndim, j - ndim)
-            self.assertEqual(a, x)
+def compare_pos_neg_dim(name, tensor_arg, arg_constr, types):
+    def tmp(self):
+        if isinstance(tensor_arg, list):
+            assert 0 not in types and 1 not in types
+            x = []
+            for arg in tensor_arg:
+                x.append(torch.randn(*arg))
+                ndim = len(arg)
+        else:
+            x = torch.randn(*tensor_arg)
+            ndim = len(tensor_arg)
+
+        n_dim_to_test = 0
+        tmp_arg = arg_constr()
+        for el in tmp_arg:
+            if el is None:
+                n_dim_to_test += 1
+
+        for dims_val in combinations(range(ndim), n_dim_to_test):
+            arg = arg_constr()
+            arg_neg = copy.copy(arg)
+            idx = 0
+            for i, v in enumerate(arg):
+                if v is None:
+                    arg[i] = dims_val[idx]
+                    arg_neg[i] = dims_val[idx] - ndim
+                    idx += 1
+
+            if 0 in types:
+                a = getattr(x, name)(*arg)
+                b = getattr(x, name)(*arg_neg)
+                self.assertEqual(a, b)
+
+            if 1 in types:
+                a = x.clone()
+                getattr(a, name + '_')(*arg)
+                b = x.clone()
+                getattr(b, name + '_')(*arg_neg)
+                self.assertEqual(a, b)
+
+            if 2 in types:
+                a = getattr(torch, name)(x, *arg)
+                b = getattr(torch, name)(x, *arg_neg)
+                self.assertEqual(a, b)
+
+    return tmp
+
+
+def idx_tensor(size, max_val):
+    return torch.LongTensor(*size).random_() % max_val
+
+# 4th argument is the supported types of call:
+# 0: method called as tensor.name(...)
+# 1: method called as tensor.name_(...)
+# 2: method called as torch.name(tensor, ...)
```

Comment:
Can you define these as variables? e.g.
```python
METHOD = 1
INPLACE_METHOD = 2
FUNCTIONAL = 4
DIM_ARG = None
```
Then you can do `METHOD | INPLACE_METHOD` to denote that you want to test both.